airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/destination/async/buffers/BufferDequeue.kt

```diff
@@ -63,7 +63,7 @@ class BufferDequeue(
 
                 // otherwise pull records until we hit the memory limit.
                 val newSize: Long = (memoryItem.size) + bytesRead.get()
-                if (newSize <= optimalBytesToRead) {
+                if (newSize <= optimalBytesToRead || output.isEmpty()) {
```

Comment:
What is this accomplishing? Is this because other changes caused `optimalBytesToRead` to be zero?

Reply:
the problem here was if `bytesRead == 0 && memoryItem.size > optimalBytesToRead`, then we never add anything to the queue. So here, regardless of `memoryItem.size` or `optimalBytesToRead`, if there's no item in the queue, we add the current one. 
For `fileTransfer` we set `optimalBytesToRead` to `1` so that we force a flush for each message. but with such a small value, any message is bigger than the optimal size, which causes an infinite loop.
Note that the infinite loop could also happen if `memoryItem.size()` was big enough and `optimalBytesToRead` was small enough. With our current settings, I don't believe it's possible, but it's just a couple of setting tweaks away...